wandb/sdk/wandb_run.py

```diff
@@ -147,14 +147,42 @@ class RunStatusChecker(object):
     For now, we just use this to figure out if the user has requested a stop.
     """
 
-    def __init__(self, interface: BackendSender, polling_interval: int = 15) -> None:
+    def __init__(
+        self,
+        interface: BackendSender,
+        stop_polling_interval: int = 15,
+        retry_polling_interval: int = 1,
+    ) -> None:
         self._interface = interface
-        self._polling_interval = polling_interval
+        self._stop_polling_interval = stop_polling_interval
+        self._retry_polling_interval = retry_polling_interval
 
         self._join_event = threading.Event()
-        self._thread = threading.Thread(target=self.check_status)
-        self._thread.daemon = True
-        self._thread.start()
+        self._stop_thread = threading.Thread(target=self.check_status)
+        self._stop_thread.daemon = True
+        self._stop_thread.start()
+
+        self._retry_thread = threading.Thread(target=self.check_retries)
+        self._retry_thread.daemon = True
+        self._retry_thread.start()
+
+    def check_retries(self) -> None:
+        join_requested = False
+        while not join_requested:
+            status_response = self._interface.communicate_status(
+                check_stop_req=False, check_retries=True
+            )
+            if status_response and status_response.retry_responses:
+                for hr in status_response.retry_responses:
+                    if (
+                        hr.http_status_code == 200 or hr.http_status_code == 0
+                    ):  # we use 0 for non-http errors (eg wandb errors)
+                        wandb.termlog(f"{hr.http_response_text}")
```

Comment:
fstrings not allowed (not available in python3.5 and below)

this also might be a bit too verbose, doing a terminal print for every retry? seems like we should be tracking status instead about what type of errors are seen and printing a termlog every N seconds?

Reply:
Maybe have print all network messages as an internal setting so we can see all the warnings if we want to debug what is going on?  

Originally i figured that we would just be keeping track of stats in the sender process and this status check would just get these stats but the approach you did is a bit heavier weight but gives us better access to the actual errors.

Reply:
so i actually didn't realize this at first but on line 113 of retry.py we only load the first retry failure (`self._num_iter == 2`) into the queue. in any case we only print at the same frequency as errors were already getting termlogged already before, there shouldn't be a visible change in frequency on the user end